conformance_tests/test_tiovx/test_bidir_ext_framework.c

```diff
@@ -0,0 +1,928 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "test_engine/test.h"
+#include "test_tiovx/test_tiovx.h"
+
+#include <VX/vx.h>
+#include <TI/tivx.h>
+#include <TI/tivx_task.h>
+#include <VX/vx_khr_pipelining.h>
+
+enum user_library_e
+{
+    USER_LIBRARY_EXAMPLE = 1
+};
+
+enum user_kernel_e
+{
+  MY_USER_KERNEL = VX_KERNEL_BASE( VX_ID_USER, USER_LIBRARY_EXAMPLE ) + 0x001
+};
+
+TESTCASE(bpExtFramework, CT_VXContext, ct_setup_vx_context, 0)
+
+/* Write some data on an image */
+void writeImage(vx_image image)
+{
+    vx_map_id id;
+    vx_rectangle_t rect = {.start_x = 0, .end_x = 1, .start_y = 0, .end_y = 2};
+    vx_imagepatch_addressing_t ipa;
+    void *ptr;
+    VX_CALL(vxMapImagePatch(image, &rect, 0, &id, &ipa, &ptr, VX_READ_AND_WRITE, VX_MEMORY_TYPE_HOST, 0));
+    *(vx_uint8 *)vxFormatImagePatchAddress2d(ptr, 0, 0, &ipa) = 1;
+    *(vx_uint8 *)vxFormatImagePatchAddress2d(ptr, 0, 1, &ipa) = 2;
+    VX_CALL(vxUnmapImagePatch(image, id));
+}
+
+/* Check data in an image */
+vx_status checkImage(vx_image image, vx_uint8 a, vx_uint8 b)
+{
+    vx_status status = (vx_status)VX_SUCCESS;
+    vx_map_id id;
+    vx_rectangle_t rect = {.start_x = 0, .end_x = 1, .start_y = 0, .end_y = 2};
+    vx_imagepatch_addressing_t ipa;
+    void *ptr;
+    status = vxMapImagePatch(image, &rect, 0, &id, &ipa, &ptr, VX_READ_AND_WRITE, VX_MEMORY_TYPE_HOST, 0);
+    vx_uint8 aa = *(vx_uint8 *)vxFormatImagePatchAddress2d(ptr, 0, 0, &ipa);
+    vx_uint8 bb = *(vx_uint8 *)vxFormatImagePatchAddress2d(ptr, 0, 1, &ipa);
+    status = vxUnmapImagePatch(image, id);
+    if (aa == a && bb == b)
+    {
+        status = (vx_status)VX_SUCCESS;
+    }
+    else
+    {
+        status = (vx_status)VX_FAILURE;
+    }
+    return status;
+}
+
+vx_status myKernelFunction(vx_node node, const vx_reference parameters[], vx_uint32 nparams)
+{
+    /* Implement a very simple operation on two of the pixels so that we can verify
+    that things have happened as they should, in the order they should. We will perform
+    these operations:
+    bidir(0,0) = bidir(0, 0) + input(0,0) + 1
+    bidir(0,1) = bidir(0, 1) + input(0,1) + 2
+    */
+    vx_status status = (vx_status)VX_SUCCESS;
+    vx_image input = (vx_image)parameters[0];
+    vx_image bidir = (vx_image)parameters[1];
+    vx_map_id id_in, id_bid;
+    vx_rectangle_t rect = {.start_x = 0, .end_x = 1, .start_y = 0, .end_y = 2};
+    vx_imagepatch_addressing_t ipa_in, ipa_bid;
+    void *ptr_in, *ptr_bid;
+    tivxTaskWaitMsecs(100); /* Slow this kernel down so anything executing in parallel may complete */
+    status = vxMapImagePatch(input, &rect, 0, &id_in, &ipa_in, &ptr_in, VX_READ_ONLY, VX_MEMORY_TYPE_HOST, 0);
+    status = vxMapImagePatch(bidir, &rect, 0, &id_bid, &ipa_bid, &ptr_bid, VX_READ_AND_WRITE, VX_MEMORY_TYPE_HOST, 0);
```

Comment:
status is returned, but not checked.  Better to use VX_CALL() which checks the status and asserts error if found.

Reply:
you can't use the VX_CALL as I explained you in my previous post. The VX_CALL can return a void.
the status is checked later in the code when we use the function